includes/type/object/class-root-query.php

```diff
@@ -0,0 +1,677 @@
+<?php
+/**
+ * Registers WooCommerce fields on the RootQuery object.
+ *
+ * @package WPGraphQL\WooCommerce\Type\WPObject
+ * @since   0.6.0
+ */
+
+namespace WPGraphQL\WooCommerce\Type\WPObject;
+
+use GraphQL\Error\UserError;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQLRelay\Relay;
+use WPGraphQL\AppContext;
+use WPGraphQL\WooCommerce\Data\Factory;
+
+/**
+ * Class - Root_Query
+ */
+class Root_Query {
+	/**
+	 * Return WooCommerce GraphQL queries.
+	 */
+	public static function register_fields() {
+		// Cart queries.
+		register_graphql_field(
+			'RootQuery',
+			'cart',
+			array(
+				'type'        => 'Cart',
+				'description' => __( 'The cart object', 'wp-graphql-woocommerce' ),
+				'resolve'     => function() {
+					$token_invalid = apply_filters( 'graphql_woocommerce_session_token_errors', null );
+					if ( $token_invalid ) {
+						throw new UserError( $token_invalid );
+					}
+
+					return Factory::resolve_cart();
+				},
+			)
+		);
+
+		register_graphql_field(
```

Comment:
You could also do `register_graphql_fields( 'RootQuery' $array_of_fields );`

Reply:
I know, I pretty much just copied and pasted the logic from the `type` classes.

Reply:
word. no biggie just thought I'd point it out. 😄